foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/metamodel/element/python/PythonStep.java

```diff
@@ -130,18 +135,25 @@ public Set<String> getImplImports() {
      * @return {@link #imports} that have been appropriately modified to include the correct
      * relative import paths for any references to metamodel-define records or dictionary types.
      */
-    private Set<String> getImportsWithRelativePaths(boolean isImplModule) {
-        Set<String> importsWithRelativePaths = new TreeSet<>();
+    private Set<String> getImports(boolean isImplModule) {
+        Set<String> importsSet = new TreeSet<>();
         for (String moduleImport : imports) {
             if (moduleImport.startsWith("from record.")
                     || moduleImport.startsWith("from dictionary.")) {
-                importsWithRelativePaths.add(moduleImport.replace("from ",
-                        isImplModule ? "from .." : "from ..."));
+                if(profileName.equals("data-delivery-pyspark"))
+                {
+                    importsSet.add(moduleImport.replace("from ",
+                            isImplModule ? "from .." : "from ..."));
+                }else if(profileName.equals("data-delivery-pyspark-pipeline")){
+                    String rootArtifact = rootArtifactId.replace("-", "_");
+                    importsSet.add(moduleImport.replace("from ",
+                            "from " + rootArtifact + "_data_records."));
+                }
```

Comment:
A: else? We could consider throwing an exception in the final else block, again just to fail faster and help track down bugs by signaling they're true root faster.